wgpu-types/src/lib.rs

```diff
@@ -3172,9 +3182,14 @@ impl TextureFormat {
 
             Self::Astc { .. } =>          (        noaa,      basic),
         };
+        
+        let sample_type = self.sample_type(None);
+        let mut is_filterable = 
+            sample_type == Some(TextureSampleType::Float { filterable: true });
+        if device_features.contains(Features::FLOAT32_FILTERABLE) {
+            is_filterable |= sample_type == Some(TextureSampleType::Float { filterable: false });
+        } 
 
-        let is_filterable =
-            self.sample_type(None) == Some(TextureSampleType::Float { filterable: true });
```

Comment:
Also, the formats are not blendable. Before this feature, it was always the case that filterable formats always supported blending as well but not anymore.

Reply:
> This needs to happen in `sample_type` since it's used by validation.

I need some help moving this logic into `sample_type`. To decided whether a format that is normally `unfilterable` is currently indeed `filterable`, we need the `device_features` object at hand. We cannot simply change the signature since this method is called from several different places in the codebase.

I'm also a bit confused. Since it's here  (in the current method `guaranteed_format_features()`) that we set the filterable and blendable `TextureFormatFeatureFlags`. So we need to make adjustments here too.
 
Also, for my understanding, I've tested and it  *does* work as it's written now. Is that expected or are there cases where it would not.

Reply:
> > This needs to happen in `sample_type` since it's used by validation.
> 
> I need some help moving this logic into `sample_type`. To decided whether a format that is normally `unfilterable` is currently indeed `filterable`, we need the `device_features` object at hand. We cannot simply change the signature since this method is called from several different places in the codebase.
> 
> I'm also a bit confused. Since it's here (in the current method `guaranteed_format_features()`) that we set the filterable and blendable `TextureFormatFeatureFlags`. So we need to make adjustments here too.

Passing `device_features` to `sample_type` is what I had in mind, I also looked around where `sample_type` is used and see no issue passing the set of features to it. There are a few places where it doesn't matter if the sample type is filterable or not, for those cases we can even pass it `Features::empty()`.

> Also, for my understanding, I've tested and it _does_ work as it's written now. Is that expected or are there cases where it would not.

The feature does work but validation won't work properly, I suspect this is why CI is also failing on the new `float32_filterable_without_feature` test.